### PR TITLE
Feature/#61635 log individual request object fields and support

### PR DESCRIPTION
**Pull request by arkovean**

### CODE REVIEW

File: Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            }
+
+            return metadatas;
+        }
+
+
+
+
+
+        //internal static IEnumerable<ReflectionMetadataInfo<TType>> ExtracPropertiesValues<TType>()
```
Comment (EranOfer) [inline Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs]: remove dead code

File: Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            _method = typeof(PropertiesMetadataPropertiesCache).GetMethod(nameof(Register), BindingFlags.NonPublic | BindingFlags.Instance);
+            _cache = new ConcurrentDictionary<Type, object[]>();
+
+        }
+
+        public IEnumerable<MetadataCacheParam> ParseIntoParams(object instance)
+        {
+            var type = instance.GetType();
+
+            if (type.IsClass == false)
```
Comment (EranOfer) [inline Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs]: Let talk about the exception,  it should not fail the request.
We should add validation in the service startup instead.

File: Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using Gigya.ServiceContract.Attributes;
+
+namespace Gigya.Microdot.SharedLogic.Events
+{
+    //public class ReflectionMetadataInfo<TType> where TType : class
+    //{
```
Comment (EranOfer) [inline Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs]: remove dead code

File: Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+            Register(instance, type);
+            var result = Resolve(instance, type);
+
+            return result;
+        }
+
+        private void Register(object instance, Type type)
+        {
+            _cache.GetOrAdd(type, x => ExtracPropertiesValues(instance, type).Cast<object>().ToArray());
```
Comment (EranOfer) [inline Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs]: you are getting the value twice in Register and in resolve let talk about it

File: Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+                };
+            }
+        }
+
+        internal static IEnumerable<ReflectionMetadataInfo> ExtracPropertiesValues(object instance, Type type)
+        {
+            var getters = type.GetProperties(BindingFlags.Public | BindingFlags.Instance).Where(p => p.CanRead)
+                .Select(x => new
+                {
+                    Getter = x.GetGetMethod(),
```
Comment (EranOfer) [inline Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs]: What happened if a getter throws an exception?
We should add try catch and write  log when you run the exception

File: Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+        private IEnumerable<MetadataCacheParam> Resolve(object instance, Type type)
+        {
+            foreach (var item in _cache[type])
+            {
+                var workinItem = (ReflectionMetadataInfo)item;
+
+                yield return new MetadataCacheParam
+                {
+                    Name = workinItem.PropertyName,
+                    Value = workinItem.ValueExtractor(instance),
```
Comment (EranOfer) [inline Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs]: What happened if a getter throws an exception?
We should add try catch and write  log when you run the exception

File: Gigya.Microdot.Orleans.Ninject.Host/MicrodotOrleansServiceHost.cs [hunk 7]
```diff
@@ -53,25 +54,32 @@ public abstract class MicrodotOrleansServiceHost : ServiceHostBase
 
         public abstract ILoggingModule GetLoggingModule();
 
+
+        protected MicrodotOrleansServiceHost()
```
Comment (EranOfer) [inline Gigya.Microdot.Orleans.Ninject.Host/MicrodotOrleansServiceHost.cs]: remove

File: Gigya.Microdot.Orleans.Ninject.Host/MicrodotOrleansServiceHost.cs [hunk 8]
```diff
@@ -84,6 +92,8 @@ protected override void OnStart()
         /// <param name="kernel"></param>
         protected virtual void PreInitialize(IKernel kernel)
         {
+             kernel.Get<ServiceValidator>()?.Validate();
+            //Validator.Validate();
```
Comment (EranOfer) [inline Gigya.Microdot.Orleans.Ninject.Host/MicrodotOrleansServiceHost.cs]: remove comment

File: Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+            var result = ExtracParams(instance, type);
+
+            return result;
+        }
+
+
+
+        private IEnumerable<MetadataCacheParam> ExtracParams(object instance, Type type)
+        {
+            var propertyMetadata = _propertyMetadataCache.GetOrAdd(type, x => ExtracPropertiesMetadata(instance, type).ToArray());
```
Comment (EranOfer) [inline Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs]: remove ToArray

File: tests/Gigya.Microdot.Orleans.Hosting.UnitTests/ReflectionMetaDataExtensionTests.cs [hunk 10]
```diff
@@ -0,0 +1,361 @@
...
+            var attribute = GetAttribute<TAttribute>(mock, propName);
+            object tmpValue;
+
+            if (TryGetValue(mock, propName, out tmpValue))
+            {
+                value.ShouldBe(value);
+            }
+            else
+            {
+                //value.ToString().StartsWith(PropertiesMetadataPropertiesCache.ErrorMessage).ShouldBeTrue();
```
Comment (EranOfer) [inline tests/Gigya.Microdot.Orleans.Hosting.UnitTests/ReflectionMetaDataExtensionTests.cs]: remove dead code

File: Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+            {
+                object value = null;
+
+                try
+                {
+                    value = item.ValueExtractor(instance);
+                }
+                catch (Exception ex)
+                {
+                    _log.Warn(string.Format(ErrorMessage, item.PropertyName));
```
Comment (EranOfer) [inline Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs]: use Tag!! instead

File: Sample/CalculatorService/CalculatorServiceHost.cs [hunk 12]
```diff
@@ -19,8 +21,10 @@ static void Main(string[] args)
             Environment.SetEnvironmentVariable("DC", "global");
             Environment.SetEnvironmentVariable("ENV", "dev");
 
+
             try
             {
+                //var serviceValidator = new ServiceValidator((new List<IValidator>()).ToArray());
```
Comment (EranOfer) [inline Sample/CalculatorService/CalculatorServiceHost.cs]: remove dead code

File: Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs (hunk 5 above)
Comment (arkovean) [reply Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs]: Done.

File: Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs (hunk 9 above)
Comment (arkovean) [reply Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs]: Array has to stay due to 
        private readonly ConcurrentDictionary<Type, ReflectionMetadataInfo[]> _propertyMetadataCache;

File: Sample/CalculatorService/CalculatorServiceHost.cs (hunk 12 above)
Comment (arkovean) [reply Sample/CalculatorService/CalculatorServiceHost.cs]: Done.

File: Gigya.Microdot.Orleans.Ninject.Host/MicrodotOrleansServiceHost.cs (hunk 7 above)
Comment (arkovean) [reply Gigya.Microdot.Orleans.Ninject.Host/MicrodotOrleansServiceHost.cs]: Done.

File: Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs (hunk 11 above)
Comment (arkovean) [reply Gigya.Microdot.SharedLogic/Events/MetadataPropertiesCache.cs]: Done (Under @EranOfer supervision

**Pull request merged**